src/main/java/com/amplitude/Amplitude.java

```diff
@@ -88,83 +76,18 @@ public synchronized void flushEvents() {
             List<Event> eventsInTransit = new ArrayList<>(eventsToSend);
             eventsToSend.clear();
             CompletableFuture.supplyAsync(() -> {
-                int responseCode = syncHttpCallWithEventsBuffer(eventsInTransit);
-                if (responseCode >= Constants.HTTP_STATUS_MIN_RETRY && responseCode <= Constants.HTTP_STATUS_MAX_RETRY) {
-                    eventsToSend.addAll(eventsInTransit);
-                    tryToFlushEventsIfNotFlushing();
+                Response response = Response.syncHttpCallWithEventsBuffer(eventsInTransit, apiKey);
+                int responseCode = response.code;
+                Status status = Response.getCodeStatus(responseCode);
+                if (status == Status.INVALID ||
+                    status == Status.PAYLOAD_TOO_LARGE ||
```

Comment:
nit: could be a helper method like `shouldRetryForStatus`, to isolate the status check a bit